checker.c

```diff
@@ -1,21 +1,53 @@
 #include <stdio.h>
 #include <assert.h>
+int TempAlerter = 0;
+int SOCAlerter = 0;
+int ChargeRateAlerter = 0;
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
+ void CheckIfTempExceedsRange(float temperature)
+  {
+      if(temperature < 0 || temperature > 45)
```

Comment:
Similar range checking duplicate code can be reduced and refactored to a single method. for ex. IsValueInRange(...)

Reply:
done